torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -1345,6 +1364,40 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
             ++af.pc;
             break;
           }
+          case TYPECHECK: {
+            int num_inputs = stack.back().toInt(), i = 0;
+            pop(stack);
+            TORCH_INTERNAL_ASSERT(stack.size() >= num_inputs);
+            if (num_inputs == 0) {
+              push(stack, true);
+              break;
+            }
+            // Check every input's shape against profiled (expected) shape.
+            for (i = 0; i < num_inputs; i++) {
+              auto& input = peek(stack, i, num_inputs);
+              TORCH_INTERNAL_ASSERT(input.isTensor());
+              auto t = input.toTensor();
+              const TypePtr& expected = af.types[inst.X];
```

Comment:
We should be comparing the actual type of I-th input with the expected size specified through the I-th output. What we are doing now is that we're comparing input types vs the expected type of the 1st output - that's probably the reason why TypeCheck almost always returns False.

Reply:
Correct! I did not handle properly the types in the interpreter, my bad. First time working on it